### PR TITLE
feat: [CDS-85968]: add support for approval id in harness approval step

### PR DESCRIPTION
**Pull request by RishabhGupta34**
## Describe your changes
Add support for harness approval id in approval step, to be used as another identifier for users to verify approval steps uniquely.
This is only step parameter changes, will add logic as part of another pr
Schema PR: https://github.com/harness/harness-schema/pull/244

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56334)

### CODE REVIEW

**Comment by RishabhGupta34** [comment]
trigger uts

**Review by abhinavmittal-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56334)* status: 2 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/steps/approval/step/harness/HarnessApprovalStepInfo.java` line 55 at r1](https://reviewable.io/reviews/harness/harness-core/56334#-Nktnrm647Crt91BygSP:-Nktnrm647Crt91BygSQ:bhndl5z) ([raw file](https://github.com/harness/harness-core/blob/354733bbec2dc0f94f64c166a766f9016a733fda/pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/steps/approval/step/harness/HarnessApprovalStepInfo.java#L55)):*
> ```Java
>   @NotEmpty @ApiModelProperty(dataType = SwaggerConstants.STRING_CLASSPATH) ParameterField<String> approvalMessage;
> 
>   @ApiModelProperty(dataType = SwaggerConstants.STRING_CLASSPATH) ParameterField<String> approvalId;
> ```

Make the changes in schema repo too

**Review by abhinavmittal-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56334)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Comment by RishabhGupta34** [comment]
trigger uts

**Review by naman-goenka — DISMISSED** [review]

**Review by naman-goenka — COMMENTED** [review]
Shouldn't we rename approvalId it usually depicts a unique mandatory identifier.

File: pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/steps/approval/step/harness/HarnessApprovalSpecParameters.java [hunk 1]
```diff
@@ -33,6 +33,7 @@
 @RecasterAlias("io.harness.steps.approval.step.harness.HarnessApprovalSpecParameters")
 public class HarnessApprovalSpecParameters implements SpecParameters {
   @NotNull ParameterField<String> approvalMessage;
+  ParameterField<String> approvalId;
```
Comment (puthrayaharness) [inline pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/steps/approval/step/harness/HarnessApprovalSpecParameters.java:36]: Let's rename this field to `callbackId`.

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

File: pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/steps/approval/step/harness/HarnessApprovalSpecParameters.java (hunk 1 above)
Comment (RishabhGupta34) [reply pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/steps/approval/step/harness/HarnessApprovalSpecParameters.java]: done

**Comment by RishabhGupta34** [comment]
trigger uts

**Review by puthrayaharness — APPROVED** [review]

**Comment by RishabhGupta34** [comment]
trigger ut9

**Review by naman-goenka — APPROVED** [review]

**Comment by RishabhGupta34** [comment]
trigger messagecheck, review

**Pull request merged**